docs/design/auth-layer.md

```diff
@@ -0,0 +1,224 @@
+--data-urlencode 'client_id=htg-auth-layer' \
+--data-urlencode 'client_secret=XVEvWl41OqmGCicoFMrehrqWYQQgUndS'
+```
+
+**response:**
+```json
+{
+    "exp": 1709196463,
+    "iat": 1709160463,
+    "jti": "e3c20f5e-cf2d-47ce-be81-ea3edf7d1152",
+    "iss": "http://host.docker.internal:8080/realms/HederaTheGraph",
+    "aud": "account",
+    "sub": "f0fa24d6-801e-45a6-8caf-08e8bee0b387",
+    "typ": "Bearer",
+    "azp": "htg-auth-layer",
+    "session_state": "aa53330d-ff69-4ea5-8b4f-7d49c964740d",
+    "acr": "1",
+    "allowed-origins": [
+        "/*"
+    ],
+    "realm_access": {
+        "roles": [
+            "default-roles-hederathegraph",
+            "offline_access",
+            "uma_authorization"
+        ]
+    },
+    "resource_access": {
+        "htg-auth-layer": {
+            "roles": [
+                "subgraph_create",
+                "subgraph_deploy"
+            ]
+        },
+        "account": {
+            "roles": [
+                "manage-account",
+                "manage-account-links",
+                "view-profile"
+            ]
+        }
+    },
+    "scope": "email profile subgraph_access",
+    "sid": "aa53330d-ff69-4ea5-8b4f-7d49c964740d",
+    "email_verified": false,
+    "name": "Developer Example",
+    "subgraph_access": "subgraph1,subgraph2,subgraph3",
+    "preferred_username": "developer1",
+    "given_name": "Developer",
+    "family_name": "Example",
+    "email": "developer1@gmail.com",
+    "client_id": "htg-auth-layer",
+    "username": "developer1",
+    "token_type": "Bearer",
+    "active": true
+}
+```
+
+### D. Hedera-TheGraph Node
+
+This component is responsible for indexing data from the Hedera network and storing it in a database. The Hedera-TheGraph node is also responsible for serving queries from the GraphQL API. The Hedera-TheGraph node will be modified to include the authentication proxy and to validate the access token before processing requests.
```

Comment:
How will it be modified?
Should call this out as it suggests this node may differ from the default node

Reply:
This was a confucion, the node server does not need to be modified to support auth, hence the proxy on top. 😅 
Thanks for noticing.

✅ Already corrected the text